src/common/2d/v_2ddrawer.h

```diff
@@ -123,20 +125,21 @@ class F2DDrawer
 		bool useTransform;
 		DMatrix3x3 transform;
 
-		DShape2D* shape2D;
+		std::shared_ptr<DShape2DBufferInfo> shape2DBufInfo;
 		int shape2DBufIndex;
 		int shape2DIndexCount;
 		int shape2DCommandCounter;
 
 		RenderCommand()
 		{
 			memset(this, 0, sizeof(*this));
```

Comment:
This becomes wrong now. There is no guarantee that `std::shared_ptr<>` filled with zero bytes is a valid object. It may work in some implementations of C++ standard library, it may not in others. Regardless of its seemingly correct functionality in your implementation, it may break at any given moment.

Reply:
If it has indeed such sloppy semantics, it might be better to use my own RefCountedPtr implementation instead. I wrote that because I considered std::shared_ptr not robust enough for the texture manager.

Personally, I'd avoid std::shared_ptr at all costs - I found it a typical STL construct, i.e. poorly designed, dangerous to use and generally undesirable.

Reply:
@coelckers - I can certainly make the change to `RefCountedPtr`, but I've run into one issue while doing this - `RefCountedPtr` has a deleted copy constructor, so this causes `RenderCommand` to no longer be copyable which the code relies on when pushing it into arrays. I'd like to ask for what would be considered the best solution before implementing this because it's been a while since I've been in any depth into C++ copy/move semantics and I'd prefer to get a second opinion. Adding a copy constructor like `std::shared_ptr` has seems to me to be the lowest effort way but I'm not sure I'm getting it entirely right:
```c
	RefCountedPtr(const RefCountedPtr& p) : ptr(p.ptr) {
		if (ptr) ptr->IncRef();
	}
```
Would appreciate some advice before I commit this.

Reply:
`RefCountedPtr<>` is a bit strange. It has `operator=` for copying and moving, but no copy and move constructors.

Also, the signature of move version seems to be wrong. It should not have `const` to be actually moveable.
https://github.com/coelckers/gzdoom/blob/4bcea0ab783c667940008a5cab6910b7a826f08c/src/common/utility/refcounted.h#L57